src/ansi-c/ansi_c_internal_additions.cpp

```diff
@@ -221,8 +221,6 @@ void ansi_c_internal_additions(std::string &code)
     "float __CPROVER_fabsf(float x);\n"
 
     // arrays
-    // NOLINTNEXTLINE(whitespace/line_length)
-    "__CPROVER_bool __CPROVER_array_equal(const void *array1, const void *array2);\n"
     // overwrite all of *dest (possibly using nondet values), even
```

Comment:
I like the idea of removing special case passes, particularly ones that are sharing unaware BUT it seems like array_copy only captures some of the functionality of array_equals.  With array_equals I can build two arrays in different ways and then test their equality.